app/src/main/java/com/example/footprnt/Profile/EditPost.java

```diff
@@ -40,17 +40,18 @@
     EditText mEtLocation;
     Button mBtnDelete;
     Button mBtnSave;
+    Post post;
+    Util util = new Util();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.fragment_edit_post);
-        Util util = new Util();
 
         // Get post from serializable extras
         Intent intent = this.getIntent();
         Bundle bundle = intent.getExtras();
-        final Post post = (Post) bundle.getSerializable(Post.class.getSimpleName());
+        post = (Post) bundle.getSerializable(Post.class.getSimpleName());
```

Comment:
We talked about this a little last week but don't use `.class.getSimpleName()`, just use the String (it's more efficient). And since you're just using the class name as a key, just save the key as a class constant.